clang/lib/Sema/CheckExprLifetime.cpp

```diff
@@ -249,7 +254,7 @@ static void visitLocalsRetainedByReferenceBinding(IndirectLocalPath &Path,
                                                   LocalVisitor Visit);
 
 template <typename T> static bool isRecordWithAttr(QualType Type) {
-  if (auto *RD = Type->getAsCXXRecordDecl())
+  if (auto *RD = Type.getNonReferenceType()->getAsCXXRecordDecl())
```

Comment:
Can this be done in a separate PR?

Reply:
This is something required by the new annotation. This avoids a false positive for:
```cpp
vsv.push_back(getOptionalSV().value());
vsv.push_back(getOptionalMySV().value());
```

Reply:
I will remove this for now from this PR and leave the tests with a FIXME. Probably needs a separate PR to keep this PR simpler.
It is not possible to do before this PR because I do not have non-lifetime_capture_by tests for it.